taxjar/taxjar-magento2-extension
# Fix bundle product partial refunds

### Context
Credit memo line items were being calculated incorrectly for bundled products because the product type of bundled products couldn't be determined.  This change fixes that bug by loading the original order item and getting the product type from it.

### Description
This PR fixes the missing product type for bundled products.  Previously the parent bundled item was refunded regardless of which specific children were refunded.  Now the refunded child items are reported correctly.  

### Performance
This PR has no measurable impact on performance.

### Testing
1. Create an order with a bundled product.  
![Screen Shot 2020-04-01 at 5 07 21 PM](https://user-images.githubusercontent.com/44789510/78194890-83b82f80-743b-11ea-84fb-e0c9522996b1.png)


2. Create a credit memo that only refunds part of the original items.
![Screen Shot 2020-04-01 at 5 07 49 PM](https://user-images.githubusercontent.com/44789510/78194871-7b5ff480-743b-11ea-899c-456e341ca81e.png)


3. Confirm only the refunded items were reported to TaxJar
![Screen Shot 2020-04-01 at 5 07 57 PM](https://user-images.githubusercontent.com/44789510/78194898-887ce380-743b-11ea-81a1-7036297fccbd.png)




#### Versions
